nightfall-optimist/src/event-handlers/block-proposed.mjs

```diff
@@ -78,35 +80,23 @@ async function blockProposedEventHandler(data) {
             blockNumberL2: block.blockNumberL2,
             mempool: false,
             offchain: true,
+            fromBlockProposer: true,
             ...tx,
           }); // must be offchain or we'll have seen them, mempool = false
         }
         return true;
       }),
     );
 
-    // Update the nullifiers we have stored, with the blockhash. These will
-    // be deleted if the block check fails and we get a rollback.  We do this
-    // before running the block check because we want to delete the nullifiers
-    // asociated with a failed block, and we can't do that if we haven't
-    // associated them with a blockHash.
-    await stampNullifiers(
-      transactions
-        .map(tx => [
-          ...tx.nullifiers.filter(nulls => nulls !== ZERO),
-          ...tx.nullifiersFee.filter(nulls => nulls !== ZERO),
-        ])
-        .flat(Infinity),
-      block.blockHash,
-    );
     // mark transactions so that they are out of the mempool,
     // so we don't try to use them in a block which we're proposing.
-    await removeTransactionsFromMemPool(block.transactionHashes, block.blockNumberL2, timeBlockL2); // TODO is await needed?
+    await removeTransactionsFromMemPool(block.transactionHashes, block.blockNumberL2); // TODO is await needed?
+    const blockCommitments = transactions.map(t => t.commitments.filter(c => c !== ZERO)).flat();
```

Comment:
You need to take into account commitmentFee too.
```suggestion
    const blockCommitments = transactions
      .map(t => [...t.commitments, ...t.commitmentFee].filter(c => c !== ZERO))
      .flat();
```

Reply:
This PR has been pending for 26 days that I wanted to fix the merge conflicts and have a merged as soon as possible because it makes major changes to optimist and is already a very PR. Since the fees was merged before this, I have to first understand fee and then add further features such that code can e added to check all the fee commitments and fee nullifiers too for duplicates and pass in the correct isNullifierFee1 or isCommitmentFee 1 flag to challenges.sol for duplicate nullifier and nullifier commitment challenges. I will to raise this as a separate issue fix

Reply:
commitmentFee change added